catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/SubmitStandingQueryRequestImpl.java

```diff
@@ -507,28 +574,42 @@ protected DAGQueryResult getData(long queryTime) {
         QueryResultsCallable queryCallable = new QueryResultsCallable(catalogQueryRequest);
 
         try {
+          LOGGER.debug("Executing query...");
           QueryResponse queryResponse = NsiliEndpoint.getGuestSubject().execute(queryCallable);
           int numHits = (int) queryResponse.getHits();
+          LOGGER.trace("Hits received: {}", numHits);
           List<Result> results = queryResponse.getResults();
           int origResultSize = results.size();
-          results = LibraryImpl.getLatestResults(results);
+          LOGGER.trace("Query returned {} results", origResultSize);
+          results = massageResults(results);
+          int filteredResultSize = results.size();
           catalogResults.addAll(results);
           int accumResults = origResultSize + (startIndex - 1);
 
-          LOGGER.trace("Processing Result {} of {}", accumResults, numHits);
+          LOGGER.trace(
+              "Cleaned up results added to be processed: {}, total hits handled: {}",
+              filteredResultSize,
+              accumResults);
 
+          if (accumResults < numHits) {
+            moreResultsAvailOnLastQuery = true;
+            startIndex = accumResults + 1;
+          } else {
+            moreResultsAvailOnLastQuery = false;
+            startIndex = 1;
+          }
+
+          /*
+           * Don't need this, we could filter out the remaining hits, but the accumResults should
```

Comment:
i'm partial to removing vs commenting out code if it isn't needed

Reply:
Agreed. Removed it